app/models/topic_user.rb

```diff
@@ -219,26 +234,19 @@ def create_missing_record(user_id, topic_id, attrs)
           auto_track_after ||= SiteSetting.default_other_auto_track_topics_after_msecs
 
           if auto_track_after >= 0 && auto_track_after <= (attrs[:total_msecs_viewed].to_i || 0)
+            attrs[:notifications_changed_at] = DateTime.now
             attrs[:notification_level] ||= notification_levels[:tracking]
           end
         end
       end
 
-      TopicUser.create!(attrs.merge!(
-        user_id: user_id,
-        topic_id: topic_id,
-        first_visited_at: now ,
-        last_visited_at: now
-      ))
     end
 
     def track_visit!(topic_id, user_id)
       now = DateTime.now
       rows = TopicUser.where(topic_id: topic_id, user_id: user_id).update_all(last_visited_at: now)
 
-      if rows == 0
-        change(user_id, topic_id, last_visited_at: now, first_visited_at: now)
-      end
+      change(user_id, topic_id, last_visited_at: now, first_visited_at: now)
```

Comment:
first_visited_at ... now? how does that make sense, are we 10000% track_visit! is only every called once?